hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAirdropValidator.java

```diff
@@ -132,21 +131,12 @@ public void validateSemantics(
         }
     }
 
-    private boolean tokenHasNoCustomFeesPaidByReceiver(TokenID tokenId, ReadableTokenStore tokenStore) {
+    private boolean tokenHasNoRoyaltyWithFallbackFee(TokenID tokenId, ReadableTokenStore tokenStore) {
```

Comment:
Fungible tokens have no royalty fee, so you can maybe keep the check for the token type, so you can skip the whole loop, directly, if its not an NFT.

Reply:
I agree, added back the token type check